crates/miden-tx/src/host/note_mast_forest_store.rs

```diff
@@ -0,0 +1,30 @@
+use alloc::{collections::BTreeMap, sync::Arc};
+
+use miden_objects::{Digest, assembly::mast::MastForest, note::NoteScript};
+use vm_processor::MastForestStore;
+
+/// Stores the MAST forests for a set of note scripts.
+pub struct NoteMastForestStore {
```

Comment:
Seemed appropriate to have a specific MAST forest store for notes only as opposed to always pre-loading it with the MAST forests related to the transaction kernel, since the use will actually be ephemeral and not long-lived. This way, the user can maintain their own store which contains the kernel-related MASTs.